bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -500,129 +498,68 @@ public void safeRun() {
                 List<PendingAddOp> pendingAdds;
 
                 if (isClosed()) {
```

Comment:
Is it possible for isClosed() but not closing here?  If closing, drainPendingAddsToErrorOut must have already happened (and no new ones can be added due to the check in doAsyncAddEntry), so this must necessarily be a noop.  I think we should be able to check closing instead and assert that pendingAddOps is already empty.

Reply:
Yeah, but I think what I'm getting at is that whatever discovered that the ledger is actually closed or closed it should really already have called drainPendingAddsToErrorOut -- this code shouldn't be discovering that for the first time.